pkg/store/proxy.go

```diff
@@ -434,11 +434,15 @@ func startStreamSeriesSet(
 				s.warnCh.send(storepb.NewWarnSeriesResponse(errors.New(w)))
 				continue
 			}
-			select {
-			case s.recvCh <- rr.r.GetSeries():
-			case <-ctx.Done():
-				s.handleErr(errors.Wrapf(ctx.Err(), "failed to receive any data from %s", s.name), done)
-				return
+
+			if series := rr.r.GetSeries(); series != nil {
+				select {
+				case s.recvCh <- series:
+				case <-ctx.Done():
+					s.handleErr(errors.Wrapf(ctx.Err(), "failed to receive any data from %s", s.name), done)
+					return
+				}
+				continue
```

Comment:
do we need continue?

Reply:
No. It's useless as, technically, is now useless the `continue` in the "warnings branch" above. I've removed both for consistency, but if you have a different opinion I can reiterate on it.